projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html

```diff
@@ -183,29 +183,21 @@ <h6 class="igx-filter-empty__title">
                            [(ngModel)]="searchValue"/>
                 </igx-input-group>
 
-                <igx-date-picker *ngIf="selectedColumn && selectedColumn.dataType === 'date'"
-                                 mode="dropdown"
-                                 [(ngModel)]="searchValue"
-                                 [locale]="grid.locale"
-                                 [outlet]="grid.outlet">
-                    <ng-template igxDatePickerTemplate let-openDialog="openDialog" let-value="value">
-                        <igx-input-group #dropDownTarget type="box" [displayDensity]="'compact'">
-                            <input #searchValueInput
-                                    igxInput
-                                    tabindex="0"
-                                    (keydown)="invokeClick($event)"
-                                    (click)="openDialog(dropDownTarget.element.nativeElement)"
-                                    [placeholder]="grid.resourceStrings.igx_grid_filter_row_date_placeholder"
-                                    autocomplete="off"
-                                    [value]="
-                                        selectedColumn.formatter
-                                        ? (value | columnFormatter:selectedColumn.formatter)
-                                        : (value | date:selectedColumn.pipeArgs.format:undefined:grid.locale)
-                                    "
-                                    [readonly]="true"
-                                    [disabled]="!selectedColumn || !selectedCondition || (selectedColumn && selectedColumn.filters.condition(selectedCondition).isUnary)"/>
-                        </igx-input-group>
-                    </ng-template>
+                <igx-date-picker #datePicker *ngIf="selectedColumn && selectedColumn.dataType === 'date'"
+                    [(value)]="searchValue"
+                    (click)="datePicker.open()"
+                    type="box"
+                    [readOnly]="true"
+                    [displayDensity]="'compact'"
+                    [locale]="grid.locale"
+                    [outlet]="grid.outlet"
+                    [displayFormat]="selectedColumn.pipeArgs.format"
+                    [placeholder]="grid.resourceStrings.igx_grid_filter_row_date_placeholder"
+                    [disabled]="!selectedColumn || !selectedCondition
+                        || (selectedColumn && selectedColumn.filters.condition(selectedCondition).isUnary)">
+                    <!-- disable default icons -->
+                    <igx-picker-toggle></igx-picker-toggle>
+                    <igx-picker-clear></igx-picker-clear>
```

Comment:
At first this looks like a strange way to disable the default icons, but on a second thought I guess it is ok.

Reply:
Admittedly not super intuitive, thus the comment. Still beats a separate template/prop to change those and another prop to hide each IMO :)